File: Cola_Lista/ColaLista.py
class Nodo:
    def __init__(self, dato):
        self.dato = dato
        self.sig = None

frente = None
fin = None

def insertar():
    global frente, fin
    elem = int(input("\nIngrese el elemento: "))
    nuevo = Nodo(elem)

    if frente is None:
        frente = fin = nuevo
    else:
        fin.sig = nuevo
        fin = nuevo
    print("Elemento insertado correctamente.")

def eliminar():
    global frente, fin
    if frente is None:
        print("\nSUBDESBORDAMIENTO (UNDERFLOW)")
        return

    elem = frente.dato
    frente = frente.sig
    if frente is None:
        fin = None
    print("\nElemento eliminado:", elem)

def mostrar():
    if frente is None:
        print("\nLa cola está vacía.")
        return

    print("\nElementos en la cola:")
    aux = frente
    while aux:
        print(aux.dato)
        aux = aux.sig

# -------- Menú ---------
op = 0
while op != 4:
    print("\n*************** MENÚ PRINCIPAL ***************")
    print("1. Insertar un elemento")
    print("2. Eliminar un elemento")
    print("3. Mostrar la cola")
    print("4. Salir")
    op = int(input("Ingrese su opción: "))

    if op == 1: insertar()
    elif op == 2: eliminar()
    elif op == 3: mostrar()
    elif op == 4: print("\nSaliendo...")
    else: print("\nOpción inválida.")

File: Cola_Lista/ColaLista.js
const readline = require("readline").createInterface({
    input: process.stdin,
    output: process.stdout
});

class Nodo {
    constructor(dato) {
        this.dato = dato;
        this.sig = null;
    }
}

let frente = null, fin = null;

function insertar() {
    readline.question("\nIngrese el elemento: ", x => {
        let nuevo = new Nodo(parseInt(x));
        if (!frente) frente = fin = nuevo;
        else { fin.sig = nuevo; fin = nuevo; }
        console.log("Elemento insertado correctamente.");
        menu();
    });
}

function eliminar() {
    if (!frente) return console.log("\nSUBDESBORDAMIENTO (UNDERFLOW)");
    let eliminado = frente.dato;
    frente = frente.sig;
    if (!frente) fin = null;
    console.log("\nElemento eliminado:", eliminado);
}

function mostrar() {
    if (!frente) return console.log("\nLa cola está vacía.");
    console.log("\nElementos en la cola:");
    for (let i = frente; i; i = i.sig) console.log(i.dato);
}

function menu() {
    console.log("\n*************** MENÚ PRINCIPAL ***************");
    console.log("1. Insertar un elemento");
    console.log("2. Eliminar un elemento");
    console.log("3. Mostrar la cola");
    console.log("4. Salir");
    readline.question("Ingrese su opción: ", op => {
        switch (parseInt(op)) {
            case 1: return insertar();
            case 2: eliminar(); break;
            case 3: mostrar(); break;
            case 4: console.log("\nSaliendo..."); return readline.close();
            default: console.log("\nOpción inválida."); break;
        }
        menu();
    });
}

menu();
